services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/modules/mounted_fs.py

```diff
@@ -87,17 +89,25 @@ def _ensure_directories(self) -> None:
         self.disk_outputs_path  # pylint:disable= pointless-statement
         set(self.disk_state_paths())
 
-    def get_inputs_docker_volume(self) -> str:
-        return f"{self.volume_name_inputs}:{self.inputs_path}"
+    @staticmethod
+    async def _get_bind_path_from_label(label: str) -> Path:
+        volume_details = await get_volume_by_label(label=label)
+        return Path(volume_details["Mountpoint"])
+
+    async def get_inputs_docker_volume(self) -> str:
+        bind_path: Path = await self._get_bind_path_from_label(self.volume_name_inputs)
+        return f"{bind_path}:{self.inputs_path}"
 
-    def get_outputs_docker_volume(self) -> str:
-        return f"{self.volume_name_outputs}:{self.outputs_path}"
+    async def get_outputs_docker_volume(self) -> str:
+        bind_path: Path = await self._get_bind_path_from_label(self.volume_name_outputs)
+        return f"{bind_path}:{self.outputs_path}"
 
-    def get_state_paths_docker_volumes(self) -> Generator[str, None, None]:
+    async def get_state_paths_docker_volumes(self) -> AsyncGenerator[str, None]:
```

Comment:
If you return an iterator, I would rather name this ``iter_state_paths_to_docker_volumes``